plans/create.pp

```diff
@@ -74,10 +80,38 @@
   Optional[String[1]] $environment               = 'production',
   Optional[Enum['linux', 'windows']] $os_type    = 'linux',
   Optional[Provision::ProviderOptions] $provider_options = undef,
+
+  ##GCP Specific Parameters
+  Optional[String[1]] $project                    = undef,
+  Optional[String[1]] $network                    = 'default',
+  Optional[String[1]] $subnetwork                 = 'default',
+  Optional[String[1]] $subnetwork_project         = undef,
 ) {
   out::message('Starting infrastructure provisioning')
   $tf_dir = "terraform/${provider}/"
 
+  # Validate the provider options for GCP
+  if $provider == 'gcp' {
+    $google_credentials = system::env('GOOGLE_CREDENTIALS')
+    $google_application_credentials = system::env('GOOGLE_APPLICATION_CREDENTIALS')
+
+    if  $google_credentials == undef and $google_application_credentials == undef {
+      fail('GOOGLE_CREDENTIALS or GOOGLE_APPLICATION_CREDENTIALS environment variable is required for Google Cloud Provider')
+    } elsif $google_credentials {
+      $profile = $google_credentials
+    } else {
+      $profile = $google_application_credentials
+    }
+
+    if $project == undef {
```

Comment:
@Ramesh7 Idea: what if we require project for all providers and use the project name instead of the resource name for the state file name?

Reply:
Like the idea but if user wanted to use some diff name then it won't work that way, so I think enabling user to provide custom name make more sense. Open for thoughts.